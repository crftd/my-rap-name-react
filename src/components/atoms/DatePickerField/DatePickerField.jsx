import React from 'react';
import { DatePicker, MuiPickersUtilsProvider } from 'material-ui-pickers';
import DateFnsUtils from '@date-io/date-fns';

export default ({ field, form, ...other }) => {
  const currentError = form.errors[field.name];
  return (
    <MuiPickersUtilsProvider utils={DateFnsUtils}>
      <DatePicker
        style={{ width: '100%' }}
        keyboard
        name={field.name}
        value={field.value}
        format="dd/MM/yyyy"
        helperText={currentError}
        error={Boolean(currentError)}
        onError={(_, error) => form.setFieldError(field.name, error)}
        onChange={date => form.setFieldValue(field.name, date, true)}
        mask={value =>
          value
            ? [/\d/, /\d/, '/', /\d/, /\d/, '/', /\d/, /\d/, /\d/, /\d/]
            : []
        }
        {...other}
      />
    </MuiPickersUtilsProvider>
  );
};
