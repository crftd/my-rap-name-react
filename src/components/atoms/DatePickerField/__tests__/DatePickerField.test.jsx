import React from 'react';
import { shallow } from 'enzyme';
import DatePickerField from '../DatePickerField.jsx';

describe('[Snapshot] DatePickerField', () => {
  it('should render', () => {
    // Arrange
    const expectedForm = {
      errors: {},
    };
    const expectedField = {
      name: 'Birthday',
    };

    // Act
    const actualTree = shallow(<DatePickerField form={expectedForm} field={expectedField} />);
    // Assert
    expect(actualTree).toMatchSnapshot();
  });
});
