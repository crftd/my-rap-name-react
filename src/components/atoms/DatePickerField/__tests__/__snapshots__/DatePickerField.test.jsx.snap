// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Snapshot] DatePickerField should render 1`] = `
<MuiPickersUtilsProvider
  utils={[Function]}
>
  <ForwardRef
    error={false}
    format="dd/MM/yyyy"
    keyboard={true}
    mask={[Function]}
    name="Birthday"
    onChange={[Function]}
    onError={[Function]}
    style={
      Object {
        "width": "100%",
      }
    }
  />
</MuiPickersUtilsProvider>
`;
